spec/formatting.md

```diff
@@ -963,4 +964,30 @@ The _Default Bidi Strategy_ is defined as follows:
          1. Append U+2069 POP DIRECTIONAL ISOLATE to `out`.
 1. Emit `out` as the formatted output of the message.
 
+> [!NOTE]
+> A _resolved value_ is derived from an _expression_
+> together with a _formatting context_.)
+> An implementation can use
+> a representation of _resolved values_ that tracks
+> everything needed to determine the directionality
+> of the formatted string representation
+> of a _resolved value_.
+> This can be accomplished by incorporating
+> the `isolate` flag used in step 2(iii),
+> as well as the separate directionality annotation
+> (one of « `'LTR'`, `'RTL'`, `'unknown'` »),
+> into the representation of _resolved values_.
+> Each _function handler_ can have its own means
+> for determining the directionality annotation
+> on the _resolved value_ it returns.
+> Alternately, an implementation could simply
+> determine directionality
+> based on the locale.
```

Comment:
On further consideration, does this perhaps belong more appropriately in the section defining _resolved values_?

Specifically, I think we ought to expand a bit the paragraph starting on [line 133](https://github.com/unicode-org/message-format-wg/blob/a19b3df18c3802bd13c1dde077efd29a95339e7a/spec/formatting.md?plain=1#L133) with something like this:

```diff
 In a _pattern_, the _resolved value_ of an _expression_ or _markup_ is used in its _formatting_.
+To account for the _Default Bidi Strategy_,
+the _resolved value_ of each _expression_ _placeholder_
+SHOULD include information about the directionality of its formatted string representation,
+as well as information about whether its formatted representation
+requires isolation from the surrounding text.
```

With that in place, this note could be simplified quite a bit, or even left out entirely.

Reply:
This is pretty good, but I wouldn't limit it to the default bidi strategy, nor would I necessarily limit it to placeholders (other expressions might have a directionality so that the direction is commutative across assignments). Perhaps:

> The _resolved value_ of each _expression_
> SHOULD include the [string direction](https://www.w3.org/TR/i18n-glossary/#dfn-string-direction)
>  of its formatted string representation,
> as well as information about whether the message author
> wants to suppress or require isolation from the surrounding _message_ text,
> such as is found in the _Default Bidi Strategy_.

Reply:
Note that _resolved value_ is an implementation-internal thing, and is therefore not expected to have public visibility during formatting. This means that attaching any normative requirements on it ought to be explicitly connected to some spec-defined consumer of that information, i.e. the _Default Bidi Strategy_.

Reply:
I made these changes. Also modified the example `MessageValue` definition to show what the directionality methods could look like.